packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -229,15 +229,105 @@ String _generatePluralMethod(Message message, String translationForMessage) {
 
   final String comment = message.description ?? 'No description provided in @${message.resourceId}';
 
-  return pluralMethodTemplate
-    .replaceAll('@(comment)', comment)
-    .replaceAll('@(name)', message.resourceId)
-    .replaceAll('@(parameters)', parameters.join(', '))
-    .replaceAll('@(dateFormatting)', generateDateFormattingLogic(message))
-    .replaceAll('@(numberFormatting)', generateNumberFormattingLogic(message))
-    .replaceAll('@(count)', countPlaceholder.name)
-    .replaceAll('@(pluralLogicArgs)', pluralLogicArgs.join(',\n'))
-    .replaceAll('@(none)\n', '');
+  if (translationForMessage.startsWith('{') && translationForMessage.endsWith('}')) {
+    return pluralMethodTemplate
+        .replaceAll('@(comment)', comment)
+        .replaceAll('@(name)', message.resourceId)
+        .replaceAll('@(parameters)', parameters.join(', '))
+        .replaceAll('@(dateFormatting)', generateDateFormattingLogic(message))
+        .replaceAll(
```

Comment:
The parameters can go on this line

Reply:
I couldn't know exactly which line. The variables are defined according to their order in the template file.

I hope this is now fine.